letta/services/tool_manager.py

```diff
@@ -70,6 +70,10 @@ def create_tool(self, pydantic_tool: PydanticTool, actor: PydanticUser) -> Pydan
             pydantic_tool.organization_id = actor.organization_id
             tool_data = pydantic_tool.model_dump()
             tool = ToolModel(**tool_data)
+            # The description is most likely auto-generated via the json_schema,
+            # so copy it over into the top-level description field
+            if tool.description is None:
```

Comment:
@mattzh72 @sarahwooders FYI I think this was a bug in the tool manager code.

If at this point the `description` is null, I think we should grab it from the `json_schema` to set it.

Honestly, we could change the following line (`tool.json_schema.get("description", None)`) to instead check if `"description" in tool.json_schema and tool.json_schema["description"] is not None`, since it's pretty problematic if this hasn't been set properly (LLM won't understand how to use the tool).

Reply:
Sg, thanks for the catch!